src/io/jesi/backpack/env.cljc

```diff
@@ -0,0 +1,100 @@
+(ns io.jesi.backpack.env
+  (:refer-clojure :exclude [symbol])
+  #?(:cljs (:require-macros [io.jesi.backpack.env]))
+  (:require
+    [clojure.string :as str]
+    [com.rpl.specter :as sp]
+    [io.jesi.backpack.fn :refer [or-fn]]
+    [io.jesi.backpack.miscellaneous :refer [named? namespaced?]]))
+
+(defn cljs?
+  "Take the &env from a macro, and tell whether we are expanding into CLJS."
+  [env]
+  ;https://groups.google.com/d/msg/clojurescript/iBY5HaQda4A/w1lAQi9_AwsJ
+  (boolean (:ns env)))
+
+;TODO better runtime detection that doesn't require predicates
+
+(def runtimes
+  "An atom containing the runtime specific predicates. Is an array-map of predicates,
+  and their runtime keyword value. The predicate should take a macro `&env` and return a
+  boolean. The runtime keyword should be a reader conditional platform tag.
+  See https://clojure.org/guides/reader_conditionals"
+  (atom (array-map cljs? :cljs)))
+
+(defn- runtime [env]
+  (if (keyword? env)
+    env
+    (or (->> @runtimes
+             (filter (fn [[pred]]
+                       (pred env)))
+             first
+             second)
+        :default)))
+
+(defmulti converter "Runtime specific converters"
+  (fn [runtime] runtime))
+
+(defmethod converter :cljs [_]
+  ;TODO use (-> &env :ns :ns-aliases) for converting to cljs?
+  ;TODO convert `catch` clause also?
+  (fn ->cljs [sym]
+    (let [ns (namespace sym)]
+      (if (and (str/starts-with? ns "clojure.")
+               (not (contains?
+                      #{"clojure.core"
+                        "clojure.core.protocols"
+                        "clojure.core.reducers"
+                        "clojure.data"
+                        "clojure.datafy"
+                        "clojure.edn"
+                        "clojure.reflect"
+                        "clojure.set"
+                        "clojure.string"
+                        "clojure.walk"
+                        "clojure.zip"}
```

Comment:
So there's a bunch of cljs namespaces that aren't renamed... yay for inconsistency